src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/api/file.ts

```diff
@@ -0,0 +1,40 @@
+import { getClient } from "@/api/client";
+import type { Components, Paths } from "@/types/openapi";
+import ApiResultHandler, { type ApiResponse } from "@/utils/apiResultHandler";
+import axios, { type AxiosRequestConfig, type AxiosProgressEvent } from "axios";
+
+const apiResultHandler = new ApiResultHandler();
+
+const uploadFile = async (
+    fileId: string,
+    file: File,
+    fileClassification: string,
+    fileTag?: string,
+    onUploadProgress?: (progressEvent: AxiosProgressEvent) => void,
+): Promise<ApiResponse<Components.Schemas.UploadFileResponse>> => {
+    const client = await getClient();
+
+    const parameters: Paths.UploadFile.PathParameters & Paths.UploadFile.HeaderParameters = {
+        fileId,
+        "file-classification": fileClassification,
+        "file-tag": fileTag,
+    };
+
+    const formData = new FormData();
+    formData.append("file", file);
+
+    const config: AxiosRequestConfig = {
+        headers: {
+            'file-classification': fileClassification,
+            ...(fileTag && { 'file-tag': fileTag }),
+            'Content-Type': 'multipart/form-data'
+        },
+        onUploadProgress,
+    };
+
+    return apiResultHandler.execute<Components.Schemas.UploadFileResponse>(
+        client.upload_file(parameters, formData as unknown as string, config)
+    );
```

Comment:
hmm could we try using as FormData? as unknown => as string seems a bit strange?

Reply:
Conversion of type 'FormData' to type 'string' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.

When I tried to convert it directly, Vs code complained and gave me this error.